dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java

```diff
@@ -92,10 +82,28 @@ private CpuInfo() {
         cpuNSFactor = nsFactor;
     }
 
+    private synchronized void initializeMbean() {
+
+        if (osmx == null) {
+            // initialize mbean required to get CPU usage info
+            osmx = ManagementFactory.getOperatingSystemMXBean();
+            mBeanServer = ManagementFactory.getPlatformMBeanServer();
+
+            try {
+                operatingSystemMbean = new ObjectName("java.lang", "type", "OperatingSystem");
+            } catch (MalformedObjectNameException e) {
+                FFDCFilter.processException(e, getClass().getName(), "CpuInfo<init>");
+            }
+        }
+    }
+
     private double getSystemCPU() {
         final String OS_ATTRIBUTE_SYSTEM_CPU_LOAD = "SystemCpuLoad";
         double cpuUsage = -1;
 
+        if (osmx == null)
```

Comment:
changes this code to look like:
```
if (operatingSystemMbean != null) {
    if (mbeanServer == null) {
        mbeanServer = ManagementFactory.getPlatformMBeanServer();
    }
    cpuUsage = (Double) mBeanServer.getAttribute(operatingSystemMbean, OS_ATTRIBUTE_SYSTEM_CPU_LOAD);
}
```
All that code goes into the try below.
Make the same change in the other method below and put it in the try.  

Also make mbeanServer a volatile field.

Reply:
Nice code ... done